compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -334,7 +353,10 @@ impl<'tcx, Prov: Provenance> ImmTy<'tcx, Prov> {
     /// given layout.
     // Not called `offset` to avoid confusion with the trait method.
     fn offset_(&self, offset: Size, layout: TyAndLayout<'tcx>, cx: &impl HasDataLayout) -> Self {
-        debug_assert!(layout.is_sized(), "unsized immediates are not a thing");
+        // Verify that the input matches its type.
+        if cfg!(debug_assertions) {
```

Comment:
Is it worth adding a `debug_assert_matches_abi` method?

Reply:
IMO no, I'd rather not duplicate this logic.

Reply:
Wouldn't it look like this, with no logic duplication?
```rust
pub fn debug_assert_matches_abi(self, abi: Abi, cx: &impl HasDataLayout) {
    if cfg!(debug_assertions) {
        self.assert_matches_abi(abi, cx);
    }
}
```

Reply:
Ah, that's what you mean.

Didn't seem worth it to me. :shrug: